charon/src/transform/prettify_cfg.rs

```diff
@@ -0,0 +1,105 @@
+use std::mem;
+
+use crate::ids::Vector;
+use crate::llbc_ast::*;
+use crate::transform::TransformCtx;
+
+use super::ctx::LlbcPass;
+
+pub struct Transform;
+
+impl Transform {
+    fn update_statements(locals: &Vector<VarId, Var>, seq: &mut [Statement]) -> Vec<Statement> {
+        // Remove double aborts. This can happen when a function call is turned into an `Abort` by
+        // `inline_local_panic_functions`.
+        if let [Statement {
+            content: RawStatement::Abort(_),
+            ..
+        }, Statement {
+            content: second_abort @ RawStatement::Abort(_),
+            ..
+        }, ..] = seq
+        {
+            *second_abort = RawStatement::Nop;
+            return Vec::new();
+        }
+        if let [Statement {
+            content: RawStatement::Call(call),
+            ..
+        }, Statement {
+            content: second_abort @ RawStatement::Abort(_),
+            ..
+        }, ..] = seq
+            && locals[call.dest.var_id].ty.is_never()
+        {
+            *second_abort = RawStatement::Nop;
+            return Vec::new();
+        }
+
+        // Change `if x then { statements... } else { break }` into `if x then { nop } else { break }; statements...` to
+        // reduce nesting.
+        if let [Statement {
+            content: RawStatement::Switch(switch),
+            span: if_span,
+        }, ..] = seq
+        {
+            // Pick a switch branch such that all other diverges, if there is one.
+            let to_extract = {
+                let mut nondivergents = switch
+                    .iter_targets_mut()
+                    .filter(|st| st.diverges().is_none());
+                match nondivergents.next() {
+                    Some(first) => {
+                        if nondivergents.next().is_none() {
+                            // There's a single branch which doesn't diverge, we pick it.
+                            Some(first)
+                        } else {
+                            // There are multiple branches that don't diverge, we don't do
+                            // anything.
+                            None
+                        }
+                    }
+                    None => {
+                        drop(nondivergents);
+                        // All the branches diverge (this is often the case in desugared loops). We
+                        // pick a branch that diverges closest (e.g. we prefer picking `break` over
+                        // `panic`). In case of a tie, we pick the longest branch.
+                        switch
```

Comment:
Does this case really improve the code?

Reply:
It does because we often have `if foo { break 0 } else { statements...; continue 0 }`, where both branches jump but it would be more readable with the `continue` branch outside the `if`

Reply:
Indeed, it makes sense.